### PR TITLE
My_Hosting UI + tests

### PR DESCRIPTION
**Pull request by nikhen-s**
Related to issue: https://github.com/swent-sp-2024-event-radar/event-radar-app/issues/110 ==> closes https://github.com/swent-sp-2024-event-radar/event-radar-app/issues/110

## Objective
- Implement the UI upon clicking My Hostings in Bottom Navigation Bar

## Acceptance Criteria
- [x] Contains "+" Button to create new event. 
- [x] Clicking party icon in Nav Bar leads to My_Hosting UI
- [x] Created tests that verified elements are displayed correctly
- [x] Implemented HostedEventsViewModel

## Changes
- Added Tests for ViewModel and UI for more coverage
- Completely Separated the State from UI, it is now all in ViewModel
- Created a `ReusableComponents` class for code-reusability
- Added test tags to searchBar to assertDisplay
- Added Functions in EventOverviewViewModel and HostedEventsViewModel to update viewList/Tab change status.

### CODE REVIEW

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+                .testTag("logo"),
+        verticalAlignment = Alignment.CenterVertically) {
+          Image(
+              painter = painterResource(id = R.drawable.event_logo),
+              contentDescription = "Event Radar Logo",
+              modifier = Modifier.size(width = 186.dp, height = 50.dp))
+        }
+
+    TabRow(
+        selectedTabIndex = selectedTabIndex,
```
Comment (RaulinN) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:126]: We are using TabRows already in the Home screen. Maybe you could create a component out of these tabs in order not to repeat a large amount of code across the app?

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+              "Join the Community",
+              EventTicket("regular", 0.0, 10000),
+              "valerian@joytigoel.com",
+              mutableSetOf("298jhk", "jwj8223"),
+              mutableSetOf("20982e2hk", "j1ou223e8223"),
+              EventCategory.COMMUNITY,
+              "89298"))
+
+  var selectedTabIndex by remember { mutableIntStateOf(0) }
+  val context = LocalContext.current
```
Comment (RaulinN) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:106]:
```suggestion
```

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+                  top.linkTo(parent.top, margin = 32.dp)
+                  start.linkTo(parent.start, margin = 16.dp)
+                }
+                .testTag("logo"),
+        verticalAlignment = Alignment.CenterVertically) {
+          Image(
+              painter = painterResource(id = R.drawable.event_logo),
+              contentDescription = "Event Radar Logo",
+              modifier = Modifier.size(width = 186.dp, height = 50.dp))
+        }
```
Comment (RaulinN) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:123]: The logo should also be a component. Every time the logo is used on a view, there's code duplication

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt (hunk 1 above)
Comment (RaulinN) [reply app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:126]: Additionally, if you don't plan on using a second tab (there's currently one), the `TabRow` + `Tab` should be deleted entirely

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+    if (selectedTabIndex == 0) { // events I host vs events I staff?
+      EventList(
+          mockEvents,
+          Modifier.fillMaxWidth().constrainAs(eventList) {
+            top.linkTo(tabs.bottom, margin = 8.dp)
+            start.linkTo(parent.start)
+            end.linkTo(parent.end)
+          })
```
Comment (RaulinN) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:165]: Reusing `EventList` is great! I would move the `EventList` composable from the `Home` file into a `component` folder. It is cleaner this way than importing stuff from Home.kt into Hosting.kt. This might also lead to cyclic dependencies which we want to avoid at all costs

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+                }
+                .testTag("floatingActionButtons")
+                .fillMaxWidth()
+                .padding(16.dp),
+        horizontalArrangement = Arrangement.Absolute.Left,
+        verticalAlignment = Alignment.CenterVertically) {
+          ExtendedFloatingActionButton(
+              text = {
+                Text(
+                    text = "Create New Event",
```
Comment (RaulinN) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:183]: Make sure you have your strings in strings.xml file

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+        onTabSelected = { tab -> navigationActions.navigateTo(tab) },
+        tabList = TOP_LEVEL_DESTINATIONS,
+        selectedItem = TOP_LEVEL_DESTINATIONS[3],
+        modifier =
+            Modifier.testTag("bottomNavMenu").constrainAs(bottomNav) {
+              bottom.linkTo(parent.bottom)
+              start.linkTo(parent.start)
+              end.linkTo(parent.end)
+            })
+  }
```
Comment (RaulinN) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:225]: You (as a group) are duplicating a lot of code! This snippet (modulo the selectedItem index) is already present 3 times in the code
- short fix (not recommended) : Create a component that represents this bottom navigation bar that you can use from all the top level navigation screens (e.g. Home, MyHosting, Profile)
- recommended fix : you should not have to hardcode the index of the `selectedItem`. Can you think of a way to remove this information?

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+}
+
+@Composable
+fun EventList(events: List<Event>, modifier: Modifier = Modifier) {
+  LazyColumn(modifier = modifier) { items(events) { event -> EventCard(event) } }
+}
```
Comment (RaulinN) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:231]: Ahhh... Nevermind then. So the `EventList` is imported from Home, but also redefined (same for EventCard). There's a """huge""" problem of code duplication in your project guys! Pay attention to what your teammates write, and use the features already implemented instead of copy-pasting the same code in a diffeerent file!

**Comment by RaulinN** [comment]
Note, you can close the associated issue (#110 in this case) automatically when merging the PR. Make sure you use one of these special keywords in the PR description : https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

TLDR: "Related to issue: #110" ==> "closes #110" in your description

File: app/src/main/java/com/github/se/eventradar/ui/component/ReusableComponents.kt [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.eventradar.R
+import com.github.se.eventradar.model.Location
+import com.github.se.eventradar.model.event.Event
+import com.github.se.eventradar.model.event.EventCategory
+import com.github.se.eventradar.model.event.EventTicket
+import java.time.LocalDateTime
+
+fun getMockEvents(): List<Event> {
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/component/ReusableComponents.kt:46]: This is no longer necessary.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+import com.github.se.eventradar.R
+import com.github.se.eventradar.ui.BottomNavigationMenu
+import com.github.se.eventradar.ui.component.*
+import com.github.se.eventradar.ui.navigation.NavigationActions
+import com.github.se.eventradar.ui.navigation.TOP_LEVEL_DESTINATIONS
+import com.github.se.eventradar.ui.navigation.getTopLevelDestination
+
+@Composable
+fun HostingScreen(navigationActions: NavigationActions) {
+  val mockEvents = getMockEvents()
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:45]: You should be able to directly use the event repository.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+    if (viewMapOrListIndex == 0) {
+      EventList(
+          mockEvents,
+          Modifier.fillMaxWidth().constrainAs(eventList) {
+            top.linkTo(divider.bottom, margin = 8.dp)
+            start.linkTo(parent.start)
+            end.linkTo(parent.end)
+          })
+    } else {
+      // To Do (Map View)
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:78]: You should be able to implement this, Map is available as of #127.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+            Modifier.constrainAs(ref = buttons) {
+                  bottom.linkTo(bottomNav.top)
+                  centerHorizontallyTo(parent)
+                }
+                .testTag("floatingActionButtons")
+                .fillMaxWidth()
+                .padding(16.dp),
+        horizontalArrangement = Arrangement.Absolute.Left,
+        verticalAlignment = Alignment.CenterVertically) {
+          CreateEventFab(onClick = { /*TODO*/}, modifier = Modifier.testTag("createEventFab"))
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:92]: If this is not working, you should put a Toast message to notify the user.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+              onClick = {
+                viewMapOrListIndex = if (viewMapOrListIndex == 0) 1 else 0
+                viewToggleIcon =
+                    if (viewMapOrListIndex == 0) Icons.Default.Place
+                    else Icons.AutoMirrored.Filled.List
+              },
+              iconVector = viewToggleIcon)
+        }
+    BottomNavigationMenu(
+        onTabSelected = { tab -> navigationActions.navigateTo(tab) },
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:107]: nit: you can use `navigationActions::navigateTo` to avoid the lambda expression.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+              start.linkTo(parent.start)
+              end.linkTo(parent.end)
+            })
+  }
+}
+
+@Composable
+fun Title(modifier: Modifier) {
+  Text(
+      text = LocalContext.current.getString(R.string.hosted_events_title),
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:122]: Not sure, but can you use `stringResource()` instead?

File: app/src/main/java/com/github/se/eventradar/ui/navigation/NavigationActions.kt [hunk 14]
```diff
@@ -41,6 +42,16 @@ class NavigationActions(val navController: NavController) {
   }
 }
 
+fun getTopLevelDestination(targetTextId: Int): TopLevelDestination {
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/navigation/NavigationActions.kt:45]: `Route` is a better/more reliable parameter than the text id, which can change at any time.

File: app/src/main/res/drawable/map.png
Comment (jgeaso1266) [inline app/src/main/res/drawable/map.png:1]: This icon should already be in the repo.

**Review by jgeaso1266 — CHANGES_REQUESTED** [review]
In general, code is well written. I did not test on my device yet, so not sure how it looks. Some small notes.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+import com.github.se.eventradar.util.toast
+
+@Composable
+fun HostingScreen(
+    viewModel: EventsOverviewViewModel = viewModel(),
+    navigationActions: NavigationActions
+) {
+  val uiState by viewModel.uiState.collectAsState()
+  LaunchedEffect(key1 = uiState.eventList) { viewModel.getEvents() }
+  var viewMapOrListIndex by remember { mutableIntStateOf(0) }
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:58]: nit: I remember @RaulinN saying something along the lines that these types of variables should go into the view model.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+import com.github.se.eventradar.ui.map.EventMap
+import com.github.se.eventradar.ui.navigation.NavigationActions
+import com.github.se.eventradar.ui.navigation.Route
+import com.github.se.eventradar.ui.navigation.TOP_LEVEL_DESTINATIONS
+import com.github.se.eventradar.ui.navigation.getTopLevelDestination
+import com.github.se.eventradar.util.toast
+
+@Composable
+fun HostingScreen(
+    viewModel: EventsOverviewViewModel = viewModel(),
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:53]: You will want a separate view model for the Hosting screen. I have some comments later to show why.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+    navigationActions: NavigationActions
+) {
+  val uiState by viewModel.uiState.collectAsState()
+  LaunchedEffect(key1 = uiState.eventList) { viewModel.getEvents() }
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:57]: This will return all events in the `events` database, you want it to only return the events hosted by the user.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+                    {
+                      top.linkTo(logo.bottom, margin = 16.dp)
+                      start.linkTo(parent.start)
+                      end.linkTo(parent.end)
+                    }))
+    HorizontalDivider(
+        modifier = Modifier.constrainAs(divider, { top.linkTo(title.bottom, margin = 10.dp) }))
+    if (viewMapOrListIndex == 0) {
+      EventList(
+          uiState.eventList
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:83]: I see your comment, but `filteredEvent` is meant to filter the events on the homescreen, not the hosting screen.

**Review by jgeaso1266 — CHANGES_REQUESTED** [review]
Your changes look mostly ok (just a couple of nits). But I see you are using the event overview view model when you should create a separate view model for the `Hosting` screen.

File: app/src/main/java/com/github/se/eventradar/viewmodel/HostedEventsViewModel.kt
Comment (Nath-Assa) [inline app/src/main/java/com/github/se/eventradar/viewmodel/HostedEventsViewModel.kt:1]: For the HostedEventsUiState data class, I supposed the viewList field represents whether the user is in List View or Map view. However, we need to stay consistent. If we add this field in this UI state, we must also add it in EventsOverviewUiState. Additionally, would love to know your rationale behind storing this directly in the UiState rather than handling it Additionally, would love to know your rationale behind storing this directly in the UiState rather than handling it in Hosting.kt.

Comment (Nath-Assa) [inline app/src/main/java/com/github/se/eventradar/viewmodel/HostedEventsViewModel.kt:1]: getHostedEvents() is very similar to getUpcomingEvents() which led to us exceeding the duplicate code threshold. Look into a way to modularize this, if possible.

**Review by Nath-Assa — CHANGES_REQUESTED** [review]
Nice work on the code overall. However, this pull request seems quite extensive. I recommend dividing it into two separate pull requests for better manageability. Moreover, we should not proceed with merging this PR until the HostingEventsViewModel has been comprehensively tested.

File: app/src/androidTest/java/com/github/se/eventradar/hosting/HostingTest.kt
Comment (Nath-Assa) [inline app/src/androidTest/java/com/github/se/eventradar/hosting/HostingTest.kt:1]: Ideally, we should have separate test files: one for UI interactions within the androidTest directory and another specifically for the HostedEventsViewModel logic in the test directory.

File: app/src/main/java/com/github/se/eventradar/model/EventsOverviewViewModel.kt
Comment (Nath-Assa) [inline app/src/main/java/com/github/se/eventradar/model/EventsOverviewViewModel.kt:1]: This file is not up to date with main. Please update from main:
Note: Since this file has been moved to the viewmodel directory, it should just be deleted.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt
Comment (Nath-Assa) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:1]: You are directly accessing FirebaseAuth.getInstance().currentUser?.uid within the composable function. While this works, it's generally better to handle such data fetching or state management outside of the UI layer.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 19]
```diff
@@ -0,0 +1,179 @@
...
+import com.google.firebase.auth.FirebaseAuth
+
+@Composable
+fun HostingScreen(
+    viewModel: HostedEventsViewModel = hiltViewModel(),
+    navigationActions: NavigationActions
+) {
+  val uiState by viewModel.uiState.collectAsState()
+  val currentUser = FirebaseAuth.getInstance().currentUser?.uid
+  LaunchedEffect(key1 = uiState.eventList) { viewModel.getHostedEvents(currentUser.toString()) }
```
Comment (Nath-Assa) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:60]: Please ensure that the case when `currentUser` is null (no user signed in) is safely handled, this will result in "null" being passed as a string, which might not be the intended behavior. Consider adding a null check or a default behavior when the user is not signed in.

File: app/src/main/java/com/github/se/eventradar/ui/home/Home.kt [hunk 20]
```diff
@@ -126,7 +130,10 @@ fun HomeScreen(
         contentColor = MaterialTheme.colorScheme.primary) {
           Tab(
               selected = selectedTabIndex == 0,
-              onClick = { selectedTabIndex = 0 },
+              onClick = {
+                uiState.tab = Tab.BROWSE
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/home/Home.kt:134]: You will not be able to update the uiState this way, you will have to have a helper function with access to `_uiState` to change it.

File: app/src/main/java/com/github/se/eventradar/ui/home/Home.kt [hunk 21]
```diff
@@ -225,30 +236,25 @@ fun HomeScreen(
...
-    FloatingActionButton(
+    ViewToggleFab(
+        modifier =
+            Modifier.padding(16.dp).testTag("viewToggleFab").constrainAs(viewToggle) {
+              bottom.linkTo(bottomNav.top)
+              absoluteRight.linkTo(parent.absoluteRight)
+            },
         onClick = {
-          viewToggleBrowseIndex = if (viewToggleBrowseIndex == 0) 1 else 0
+          uiState.viewList = !uiState.viewList
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/home/Home.kt:253]: Same here, to change the uiState, you will need access to `_uiState` in the viewModel. I see that there's no compilation error because you made the variable mutable (using `var`). It should not be possible to directly edit the UI state of the view model (it breaks the MVVM architecture). Instead, make everything a `val` and use helper functions to modify it. See examples in the bootcamp if needed. 

I see @christinehahaa also did this for her changes, so it would be good to sync up with her as well.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 22]
```diff
@@ -0,0 +1,179 @@
...
+import com.github.se.eventradar.ui.navigation.NavigationActions
+import com.github.se.eventradar.ui.navigation.Route
+import com.github.se.eventradar.ui.navigation.TOP_LEVEL_DESTINATIONS
+import com.github.se.eventradar.ui.navigation.getTopLevelDestination
+import com.github.se.eventradar.util.toast
+import com.github.se.eventradar.viewmodel.HostedEventsViewModel
+
+@Composable
+fun HostingScreen(
+    uid: String,
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:55]: I might have been the one to recommend this, but after researching this myself, this won't work with the bottom navigation. #171 now allows for the correct user ID to be received using `FirebaseAuth.getInstance().currentUser!!.uid`.

File: app/src/main/java/com/github/se/eventradar/ui/navigation/NavGraph.kt [hunk 23]
```diff
@@ -34,8 +36,14 @@ fun NavGraph(navController: NavHostController) {
...
     }
     composable(Route.MY_HOSTING) {
-      HomeScreen(navigationActions = navActions)
-      context.toast("My hosted events screen needs to be implemented")
+      val currentUser = FirebaseAuth.getInstance().currentUser
+      val uid =
+          if (currentUser != null) {
+            currentUser.uid
+          } else {
+            throw IllegalStateException("User is not authenticated")
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/navigation/NavGraph.kt:44]: Not a good idea to throw an uncaught exception in Android code. We can use null assertion instead. I also think it's better to have this call in the viewModel than in the navigation (but if you think this is better, I'd be curious to know you reasoning).

**Review by jgeaso1266 — CHANGES_REQUESTED** [review]
A few important changes regarding the uiState, and a suggestion with the hosting screen. Also, you are at 69.2% coverage. There are only a few more lines that need to be covered looking at SonarCloud. Maybe invest a little time trying to get coverage to >=80%. Hopefully shouldn't take long.

File: app/src/main/java/com/github/se/eventradar/ui/navigation/NavGraph.kt (hunk 23 above)
Comment (nikhen-s) [reply app/src/main/java/com/github/se/eventradar/ui/navigation/NavGraph.kt:44]: I think that it's better for this call to be in viewModel too rather than NavGraph since only the eventCards rendered will depend on the userId, other parts of the Hosting UI itself should not depend on the userId, therefore it should be passed as a dependency only to the getHostedEvents function. The reason I passed it from navigation was to avoid calling user data from the HostingUI, but it didn't really solve the problem because I still called it in the NavGraph which is still part of UI layer. I think calling the user in the viewModel is a better way to do it such that it prepares all the data for the UI, instead of having the NavGraph do it. Thanks!

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt (hunk 22 above)
Comment (nikhen-s) [reply app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:55]: I tested it with the bottom navigation and it worked without errors since the bottom navigation bar navigates to different screen based on the routes, and the initialization of uid is defined in the NavController to go to the HostingScreen composable. However, it makes sense to put the uid in ViewModel instead, so I will move it there (my reply to your comment below has my full reasoning to follow your suggestion).

File: app/src/main/java/com/github/se/eventradar/viewmodel/HostedEventsViewModel.kt [hunk 24]
```diff
@@ -23,40 +25,51 @@ constructor(
...
-        is Resource.Failure -> {
-          Log.d("HostedEventsViewModel", "Error fetching user document")
-          _uiState.value =
-              _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
-        }
       }
     }
   }
+
+  fun onViewListStatusChanged(
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/viewmodel/HostedEventsViewModel.kt:67]: nit: this doesn't need to take in the boolean. Whenever it's called, we know the new value will be the opposite of what it is now.

**Review by jgeaso1266 — COMMENTED** [review]
One question and a few nits

File: app/src/main/java/com/github/se/eventradar/ui/component/ReusableComponents.kt [hunk 25]
```diff
@@ -0,0 +1,303 @@
...
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.eventradar.R
+import com.github.se.eventradar.model.event.Event
+import com.github.se.eventradar.model.event.EventCategory
+import com.github.se.eventradar.viewmodel.EventsOverviewUiState
+
+fun getIconFromViewListBool(viewList: Boolean): ImageVector {
+  if (viewList) return Icons.Default.Place else return Icons.AutoMirrored.Filled.List
```
[the remark this answers is not in the document]
Comment (nikhen-s) [reply app/src/main/java/com/github/se/eventradar/ui/component/ReusableComponents.kt:67]: Since it is both used in Hosting and Home, I thought that putting this helper function helps with the readibility. Additionally, if we wanted to change the Icons/were dissatisfied, we just need to change it in this one function and it will propagate.

File: app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt [hunk 26]
```diff
@@ -0,0 +1,165 @@
...
+      EventList(
+          uiState.eventList.filteredEvents,
+          Modifier.testTag("eventList").fillMaxWidth().constrainAs(eventList) {
+            top.linkTo(divider.bottom, margin = 8.dp)
+            start.linkTo(parent.start)
+            end.linkTo(parent.end)
+          })
+    } else {
+      EventMap(
+          uiState.eventList.allEvents,
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/hosting/Hosting.kt:87]: nit: `filteredEvents` would also be here

**Review by jgeaso1266 — COMMENTED** [review]
Go through the SonarCloud one more time and fix any "medium" maintainability issues. But after that the code should be good to go.

File: app/src/main/java/com/github/se/eventradar/ui/navigation/NavigationActions.kt [hunk 27]
```diff
@@ -41,6 +42,16 @@ class NavigationActions(val navController: NavController) {
...
 
+fun getTopLevelDestination(targetRoute: String): TopLevelDestination {
+  val foundDestination = TOP_LEVEL_DESTINATIONS.find { it.route == targetRoute }
+  if (foundDestination != null) {
+    return foundDestination
+  } else {
+    Log.d("Null Error", "Top level destination does not exist for R String Id: $targetRoute")
+    return TOP_LEVEL_DESTINATIONS[2]
+  }
+}
```
[the remark this answers is not in the document]
Comment (nikhen-s) [reply app/src/main/java/com/github/se/eventradar/ui/navigation/NavigationActions.kt]: Thank you, will remove all code smells from now on before asking for a PR Review.

**Review by jgeaso1266 — APPROVED** [review]
LGTM

**Review by Nath-Assa — APPROVED** [review]
